clients/src/test/java/org/apache/kafka/clients/consumer/internals/FetcherTest.java

```diff
@@ -897,6 +897,145 @@ public void testFetchOnPausedPartition() {
         assertTrue(client.requests().isEmpty());
     }
 
+    @Test
```

Comment:
Do we have a test case which covers the case where the user seeks to a new offset while a partition is paused? In this case, we expect the data to be discarded when the partition is resumed.

Reply:
I did a pass over `FetcherTest` and didn't see this scenario exactly.  I added another test called `testFetchDiscardedAfterPausedPartitionResumedAndSeekedToNewOffset` (a bit of a mouthful).  Does it create the scenario you were thinking?